src/borg/_hashindex.c

```diff
@@ -195,7 +195,8 @@ hashindex_lookup(HashIndex *index, const unsigned char *key, int *start_idx)
          * This is an invalid state we never should get into, see
          * upper_limit and min_empty.
          */
-        assert(idx != start);
+        if(idx == start)
+            return -2;
```

Comment:
sounds good, i pushed some new commits with that.

Reply:
btw, i thought about removing the `idx == -2` term from the check.

if all goes well, it is not needed, because the first term will already trigger the resize.
but, if that would not trigger due to wrong values, it is still useful to trigger based on -2 return value.

thus, i kept it.